routes/offerings.js

```diff
@@ -0,0 +1,126 @@
+const express = require("express");
+const router = express.Router();
+const Offering = require("../models/Offering");
+const Posting = require("../models/Posting");
+const Notifications = require("../utils/notificationsUtil");
+
+// http://localhost/api/offerings/{offeringId}/accept ///////////////////////////////////////////////////////
+// POST
+// Accepts the indicated offering
+router.post("/:offeringId/accept", async (req, res) => {
+  try {
+    const offering = await Offering.findOneAndUpdate(
+      {
+        _id: req.params.offeringId,
+      },
+      {
+        $set: { status: "ACCEPTED" },
+      },
+      {
+        new: true,
+      }
+    );
+
+    await Notifications(
+      offering.userId,
+      "offeringAccepted",
+      "Your offering has been accepted!"
+    );
+
+    const allPostingOfferings = await Posting.findOneAndUpdate(
```

Comment:
allPostingOfferings is undefined. Should split the call to the DB and the access to the offerings.

`const posting = await Posting.findOneAnUpdate(.....);`
`const offerings = posting.offerings;`